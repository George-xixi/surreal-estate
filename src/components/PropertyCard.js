import React from "react";
import PropTypes from "prop-types";

const PropertyCard = ({ details }) => {
  const { title, city, type, bedrooms, bathrooms, price, email } = details;
  const mailTo = `mailto:${email}?subject=Interested%20in${title}&body=I%20have%20seen%20your%20property%20on%20Surreal%20Estate%2C%20aPropend%20would%20like%20to%20request%20more%20information%20and%20arrange%20a%20viewing%20please.`;
  return (
    <div className="property-card">
      <h3>{title}</h3>
      <p>
        {city} - {type}
      </p>
      <p>Beds - {bedrooms}</p>
      <p>Baths - {bathrooms}</p>
      <p>Â£{price}</p>
      <a href={mailTo}>Email</a>
    </div>
  );
};

PropertyCard.propTypes = {
  details: PropTypes.shape({
    title: PropTypes.string.isRequired,
    city: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    bedrooms: PropTypes.number.isRequired,
    bathrooms: PropTypes.number.isRequired,
    price: PropTypes.number.isRequired,
    email: PropTypes.string.isRequired,
  }).isRequired,
};

export default PropertyCard;
