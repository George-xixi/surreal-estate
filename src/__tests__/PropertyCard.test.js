import React from "react";
import { render, screen } from "@testing-library/react";
import PropertyCard from "../components/PropertyCard";

describe("PropertyCard", () => {
  const validProps = {
    title: "1 Bed Flat",
    city: "Manchester",
    type: "Flat",
    bedrooms: 1,
    bathrooms: 2,
    price: 1000,
    email: "george@clubpenguin.com",
  };

  it("renders title correctly", () => {
    const { asFragment } = render(<PropertyCard details={validProps} />);
    const titleElement = screen.getByText("1 Bed Flat");

    expect(asFragment()).toMatchSnapshot();
    expect(titleElement).toBeInTheDocument();
  });

  it("renders city correctly", () => {
    const { asFragment } = render(<PropertyCard details={validProps} />);
    const cityElement = screen.getByText(/Manchester/);

    expect(asFragment()).toMatchSnapshot();
    expect(cityElement).toHaveTextContent("Manchester - Flat");
  });

  it("renders type correctly", () => {
    const { asFragment } = render(<PropertyCard details={validProps} />);
    const typeElement = screen.getByText(/ - Flat/);

    expect(asFragment()).toMatchSnapshot();
    expect(typeElement).toHaveTextContent("Manchester - Flat");
  });

  it("renders bedrooms correctly", () => {
    const { asFragment } = render(<PropertyCard details={validProps} />);
    const bedElement = screen.getByText("Beds - 1");

    expect(asFragment()).toMatchSnapshot();
    expect(bedElement).toBeInTheDocument();
  });

  it("renders bathrooms correctly", () => {
    const { asFragment } = render(<PropertyCard details={validProps} />);
    const bathElement = screen.getByText("Baths - 2");

    expect(asFragment()).toMatchSnapshot();
    expect(bathElement).toBeInTheDocument();
  });

  it("renders price correctly", () => {
    const { asFragment } = render(<PropertyCard details={validProps} />);
    const priceElement = screen.getByText(/1000/);

    expect(asFragment()).toMatchSnapshot();
    expect(priceElement).toHaveTextContent("Â£1000");
  });

  it("renders email correctly", () => {
    const { asFragment } = render(<PropertyCard details={validProps} />);
    const emailElement = screen.getByText("Email");

    expect(asFragment()).toMatchSnapshot();
    expect(emailElement).toHaveAttribute("href");
  });
});
