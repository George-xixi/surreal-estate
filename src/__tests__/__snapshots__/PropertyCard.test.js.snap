// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders bathrooms correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <h3>
      1 Bed Flat
    </h3>
    <p>
      Manchester - Flat
    </p>
    <p>
      Beds - 1
    </p>
    <p>
      Baths - 2
    </p>
    <p>
      £1000
    </p>
    <a
      href="mailto:george@clubpenguin.com?subject=Interested%20in1 Bed Flat&body=I%20have%20seen%20your%20property%20on%20Surreal%20Estate%2C%20aPropend%20would%20like%20to%20request%20more%20information%20and%20arrange%20a%20viewing%20please."
    >
      Email
    </a>
  </div>
</DocumentFragment>
`;

exports[`PropertyCard renders bedrooms correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <h3>
      1 Bed Flat
    </h3>
    <p>
      Manchester - Flat
    </p>
    <p>
      Beds - 1
    </p>
    <p>
      Baths - 2
    </p>
    <p>
      £1000
    </p>
    <a
      href="mailto:george@clubpenguin.com?subject=Interested%20in1 Bed Flat&body=I%20have%20seen%20your%20property%20on%20Surreal%20Estate%2C%20aPropend%20would%20like%20to%20request%20more%20information%20and%20arrange%20a%20viewing%20please."
    >
      Email
    </a>
  </div>
</DocumentFragment>
`;

exports[`PropertyCard renders city correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <h3>
      1 Bed Flat
    </h3>
    <p>
      Manchester - Flat
    </p>
    <p>
      Beds - 1
    </p>
    <p>
      Baths - 2
    </p>
    <p>
      £1000
    </p>
    <a
      href="mailto:george@clubpenguin.com?subject=Interested%20in1 Bed Flat&body=I%20have%20seen%20your%20property%20on%20Surreal%20Estate%2C%20aPropend%20would%20like%20to%20request%20more%20information%20and%20arrange%20a%20viewing%20please."
    >
      Email
    </a>
  </div>
</DocumentFragment>
`;

exports[`PropertyCard renders email correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <h3>
      1 Bed Flat
    </h3>
    <p>
      Manchester - Flat
    </p>
    <p>
      Beds - 1
    </p>
    <p>
      Baths - 2
    </p>
    <p>
      £1000
    </p>
    <a
      href="mailto:george@clubpenguin.com?subject=Interested%20in1 Bed Flat&body=I%20have%20seen%20your%20property%20on%20Surreal%20Estate%2C%20aPropend%20would%20like%20to%20request%20more%20information%20and%20arrange%20a%20viewing%20please."
    >
      Email
    </a>
  </div>
</DocumentFragment>
`;

exports[`PropertyCard renders price correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <h3>
      1 Bed Flat
    </h3>
    <p>
      Manchester - Flat
    </p>
    <p>
      Beds - 1
    </p>
    <p>
      Baths - 2
    </p>
    <p>
      £1000
    </p>
    <a
      href="mailto:george@clubpenguin.com?subject=Interested%20in1 Bed Flat&body=I%20have%20seen%20your%20property%20on%20Surreal%20Estate%2C%20aPropend%20would%20like%20to%20request%20more%20information%20and%20arrange%20a%20viewing%20please."
    >
      Email
    </a>
  </div>
</DocumentFragment>
`;

exports[`PropertyCard renders title correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <h3>
      1 Bed Flat
    </h3>
    <p>
      Manchester - Flat
    </p>
    <p>
      Beds - 1
    </p>
    <p>
      Baths - 2
    </p>
    <p>
      £1000
    </p>
    <a
      href="mailto:george@clubpenguin.com?subject=Interested%20in1 Bed Flat&body=I%20have%20seen%20your%20property%20on%20Surreal%20Estate%2C%20aPropend%20would%20like%20to%20request%20more%20information%20and%20arrange%20a%20viewing%20please."
    >
      Email
    </a>
  </div>
</DocumentFragment>
`;

exports[`PropertyCard renders type correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    <h3>
      1 Bed Flat
    </h3>
    <p>
      Manchester - Flat
    </p>
    <p>
      Beds - 1
    </p>
    <p>
      Baths - 2
    </p>
    <p>
      £1000
    </p>
    <a
      href="mailto:george@clubpenguin.com?subject=Interested%20in1 Bed Flat&body=I%20have%20seen%20your%20property%20on%20Surreal%20Estate%2C%20aPropend%20would%20like%20to%20request%20more%20information%20and%20arrange%20a%20viewing%20please."
    >
      Email
    </a>
  </div>
</DocumentFragment>
`;
